.github/workflows/cont_integration.yml

```
@@ -1,23 +1,21 @@
-on: [pull_request]
+on:
+  push:
+    branches:
+      - master
+  pull_request: {}
 
 name: CI
 
 jobs:
 
-  cosmetics:
-    runs-on: ubuntu-20.04
+  build:
+    strategy:
+      matrix:
+        os: ["macos-12", "ubuntu-22.04"]
```

do we specifically want these, or is latest fine? 

```suggestion
        os: ["macos-latest", "ubuntu-latest"]
```

---

I did this on purpose I suppose for personal preference, I prefer to manually upgrade it when needed so that if any error arise you know it's because of the OS version while if it's automatically done in the backgroud you may not be aware.